ApiReview/API.Polly.RateLimiting/NoDocs/Polly.RateLimiting/ResilienceRateLimiter.cs

```diff
@@ -0,0 +1,13 @@
+// Assembly 'Polly.RateLimiting'
+
+using System.Runtime.CompilerServices;
+using System.Threading.RateLimiting;
+using System.Threading.Tasks;
+
+namespace Polly.RateLimiting;
+
+public sealed class ResilienceRateLimiter
```

Comment:
this type doesn't exposae any useful methods. does it need to be public?

Reply:
It unifies limiters and partitioned limiters so we don't have to duplicate the `RateLimiterStrategy` options while still allowing more advanced scenarios. Enables the use of partitioned limiter in standard pipeline too.

Examples:

``` csharp
// simple usage
new ResiliencePipelineBuilder()
    .AddConcurrencyLimiter(10, 10)
    .Build();

// explicit usage
new ResiliencePipelineBuilder()
    .AddRateLimiter(new ConcurrencyLimiter(new ConcurrencyLimiterOptions
    {
        PermitLimit = 10,
        QueueLimit = 10
    }))
    .Build();

// using options + concurrency limiter
ConcurrencyLimiter concurrencyLimiter = new ConcurrencyLimiter(new ConcurrencyLimiterOptions
{
    PermitLimit = 10,
    QueueLimit = 10
});

new ResiliencePipelineBuilder()
    .AddRateLimiter(new RateLimiterStrategyOptions
    {
        RateLimiter = ResilienceRateLimiter.Create(concurrencyLimiter)
    })
    .Build();

// using options + partitioned limiter (advanced)
PartitionedRateLimiter<ResilienceContext> partitionedLimiter = PartitionedRateLimiter.Create<ResilienceContext, string>(
    context => RateLimitPartition.GetConcurrencyLimiter(
        GetPartionKey(context),
        key => new ConcurrencyLimiterOptions
        {
            PermitLimit = 10,
            QueueLimit = 10
        }));

new ResiliencePipelineBuilder()
    .AddRateLimiter(new RateLimiterStrategyOptions
    {
        RateLimiter = ResilienceRateLimiter.Create(partitionedLimiter)
    })
    .Build();
```